src/Directory.Build.targets

```diff
@@ -0,0 +1,46 @@
+<Project>
+
+  <!--
+    We need to use conditional compilation for some platform-specific code in certain cases. 
+    The macros declared here represent the platform of the build machine that is compiling the code.
+    You are wrong if you use these macros in code that compiles into a multi-platform binary.
+    (If we ship one binary for all 3 platforms, there will only be one build machine platform and it won't match the 2 others.)
+  -->
+  <PropertyGroup Condition=" $(MSBuildProjectExtension) == '.csproj' ">
+    <IsWindowsBuild Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Windows)))' == 'true'">true</IsWindowsBuild> 
+    <IsOSXBuild Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::OSX)))' == 'true'">true</IsOSXBuild> 
+    <IsLinuxBuild Condition="'$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Linux)))' == 'true'">true</IsLinuxBuild>
+
+    <DefineConstants Condition=" $(IsLinuxBuild) == 'true' ">$(DefineConstants);LINUX_BUILD</DefineConstants>
+    <DefineConstants Condition=" $(IsOSXBuild) == 'true' ">$(DefineConstants);OSX_BUILD</DefineConstants>
+    <DefineConstants Condition=" $(IsWindowsBuild) == 'true' ">$(DefineConstants);WINDOWS_BUILD</DefineConstants>
+  </PropertyGroup>
+
+  <!-- Nuget Publishing related properties that are common to all packages -->
+  <PropertyGroup Condition="'$(EnableNuget)' == 'true'" >
+    <!-- <GenerateNuspecDependsOn>$(GenerateNuspecDependsOn);</GenerateNuspecDependsOn> -->
+    <PackageOutputPath>$(MSBuildThisFileDirectory)\..\out</PackageOutputPath>
+    <Authors>Microsoft</Authors>
+    <Owners><!-- Don't set this field --></Owners>
+    <PackageProjectUrl>https://github.com/Azure/bicep-types-az</PackageProjectUrl>
```

Comment:
Should this be `https://github.com/Azure/bicep-types`

Reply:
Yup, good catch!

Reply:
Fixed in #8